### PR TITLE
Added bounds for the inverse NTT.

### PR DESCRIPTION
**Pull request by xvzcf**

### CODE REVIEW

File: src/kem/kyber/constant_time_ops.rs [hunk 1]
```diff
@@ -5,8 +5,12 @@ use super::constants::SHARED_SECRET_SIZE;
 
 #[inline]
 fn is_non_zero(value: u8) -> u8 {
```
Comment (franziskuskiefer) [inline src/kem/kyber/constant_time_ops.rs:7]: This is a function that you could test well with quickcheck or proptest. How about adding such a test for it?
Also, it's a good target for a small, easy F* proof.
But you could do that in a follow up.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks! Generally looks good. Only a couple small comments

File: fuzz/fuzz_targets/kyber1024_implicit_rejection.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), rand::Error> {
+        self.fill_bytes(dest);
+        Ok(())
+    }
+}
+
+impl CryptoRng for FuzzRng {}
+
+fn modify_ciphertext(alg: Algorithm, rng: &mut (impl CryptoRng + Rng), ciphertext: Ct) -> Ct {
```
[the remark this answers is not in the document]
Comment (xvzcf) [reply fuzz/fuzz_targets/kyber1024_implicit_rejection.rs:38]: Opened https://github.com/cryspen/libcrux/issues/129 to track this.

File: src/kem/kyber/ntt.rs [hunk 3]
```diff
@@ -110,36 +110,70 @@ pub(in crate::kem::kyber) fn ntt_vector_u<const VECTOR_U_COMPRESSION_FACTOR: usi
...
+                .take($parameter)
+                .all(|coefficient| coefficient.abs() < (128 / $parameter) * FIELD_MODULUS));
+        };
+    }
 
+    debug_assert!(
+        re.coefficients[0].abs() < 128 * coefficient_bound
+            && re.coefficients[1].abs() < 128 * coefficient_bound
+    );
+    bound_coefficient_set!(2);
```
[the remark this answers is not in the document]
Comment (xvzcf) [reply src/kem/kyber/ntt.rs:166]: How about `bound_for_set_of_coefficients`? A bit wordy but less ambiguous.

File: src/kem/kyber/constant_time_ops.rs (hunk 1 above)
Comment (xvzcf) [reply src/kem/kyber/constant_time_ops.rs:11]: Added some hax attributes as discussed in the call.

File: src/kem/kyber/ntt.rs [hunk 4]
```diff
@@ -110,36 +110,70 @@ pub(in crate::kem::kyber) fn ntt_vector_u<const VECTOR_U_COMPRESSION_FACTOR: usi
 }
 
 #[inline(always)]
-fn invert_ntt_montgomery(mut re: KyberPolynomialRingElement) -> KyberPolynomialRingElement {
+fn invert_ntt_montgomery<const K: usize>(
+    mut re: KyberPolynomialRingElement,
+) -> KyberPolynomialRingElement {
+    // We only ever call this function after matrix/vector multiplication
+    let coefficient_bound = (K as i32) * FIELD_MODULUS;
```
[the remark this answers is not in the document]
Comment (xvzcf) [reply src/kem/kyber/ntt.rs]: Done.

**Review by franziskuskiefer — APPROVED** [review]
Thanks. Let's get this in.
